src/generic/numberparsing.h

```diff
@@ -1,80 +1,1145 @@
+                                 0xdd94b7868e94050a,
+                                 0xca7cf2b4191c8326,
+                                 0xfd1c2f611f63a3f0,
+                                 0xbc633b39673c8cec,
+                                 0xd5be0503e085d813,
+                                 0x4b2d8644d8a74e18,
+                                 0xddf8e7d60ed1219e,
+                                 0xcabb90e5c942b503,
+                                 0x3d6a751f3b936243,
+                                 0xcc512670a783ad4,
+                                 0x27fb2b80668b24c5,
+                                 0xb1f9f660802dedf6,
+                                 0x5e7873f8a0396973,
+                                 0xdb0b487b6423e1e8,
+                                 0x91ce1a9a3d2cda62,
+                                 0x7641a140cc7810fb,
+                                 0xa9e904c87fcb0a9d,
+                                 0x546345fa9fbdcd44,
+                                 0xa97c177947ad4095,
+                                 0x49ed8eabcccc485d,
+                                 0x5c68f256bfff5a74,
+                                 0x73832eec6fff3111,
+                                 0xc831fd53c5ff7eab,
+                                 0xba3e7ca8b77f5e55,
+                                 0x28ce1bd2e55f35eb,
+                                 0x7980d163cf5b81b3,
+                                 0xd7e105bcc332621f,
+                                 0x8dd9472bf3fefaa7,
+                                 0xb14f98f6f0feb951,
+                                 0x6ed1bf9a569f33d3,
+                                 0xa862f80ec4700c8,
+                                 0xcd27bb612758c0fa,
+                                 0x8038d51cb897789c,
+                                 0xe0470a63e6bd56c3,
+                                 0x1858ccfce06cac74,
+                                 0xf37801e0c43ebc8,
+                                 0xd30560258f54e6ba,
+                                 0x47c6b82ef32a2069,
+                                 0x4cdc331d57fa5441,
+                                 0xe0133fe4adf8e952,
+                                 0x58180fddd97723a6,
+                                 0x570f09eaa7ea7648};
+
+// Attempts to compute i * 10^(power) exactly; and if "negative" is
+// true, negate the result.
+// This function will only work in some cases, when it does not work, success is
+// set to false. This should work *most of the time* (like 99% of the time).
+// We assume that power is in the [FASTFLOAT_SMALLEST_POWER,
+// FASTFLOAT_LARGEST_POWER] interval: the caller is responsible for this check.
+// We assume that i is non-zero: the caller is responsible for this check.
+really_inline double compute_float_64(int64_t power, uint64_t i, bool negative,
+                                      bool *success) {
+  // we start with a fast path
+  if (-22 <= power && power <= 22 && i <= 9007199254740991) {
```

Comment:
there are some more fast path cases in disguise, see http://www.exploringbinary.com/fast-path-decimal-to-floating-point-conversion/ and line 117 of https://rapidjson.org/strtod_8h_source.html

Reply:
> there are some more fast path cases in disguise

I know, but it does not help performance.

Reply:
I will document it in comments.

Reply:
You need  22 < power *and* power <  22 + 16 *and* (i * 10^(x-22) < 2^53) for this second fast path to work. If you you have 22 < power *and* power <  22 + 16, and then you optimistically compute "i * 10^(x-22)", there is still a chance that you have wasted your time if i * 10^(x-22) > 2^53. It makes the use cases of this optimization uncommon.